src/Components/Components/src/EventCallbackFactoryBinderExtensions.cs

```diff
@@ -642,5 +697,139 @@ private static DateTime ConvertDateTime(object obj, string format)
+                    {
+                        converter = ConvertToDateTime;
+                    }
+                    else if (typeof(T) == typeof(DateTime?))
+                    {
+                        converter = ConvertToNullableDateTime;
+                    }
+                    else if (typeof(T).IsEnum)
+                    {
+                        // We have to deal invoke this dynamically to work around the type constraint on Enum.TryParse.
+                        var method = _convertToEnum ??= typeof(EventCallbackFactoryBinderExtensions).GetMethod(nameof(ConvertToEnum), BindingFlags.NonPublic | BindingFlags.Static);
+                        converter = method.MakeGenericMethod(typeof(T)).CreateDelegate(typeof(BindConverter<T>), target: null);
+                    }
+                    else if (Nullable.GetUnderlyingType(typeof(T)) is Type innerType && innerType.IsEnum)
+                    {
+                        // We have to deal invoke this dynamically to work around the type constraint on Enum.TryParse.
+                        var method = _convertToNullableEnum ??= typeof(EventCallbackFactoryBinderExtensions).GetMethod(nameof(ConvertToNullableEnum), BindingFlags.NonPublic | BindingFlags.Static);
+                        converter = method.MakeGenericMethod(innerType).CreateDelegate(typeof(BindConverter<T>), target: null);
+                    }
+                    else
+                    {
+                       converter = MakeTypeConverterConverter<T>();
+                    }
+
+                    _cache.TryAdd(typeof(T), converter);
+                }
+
+                return (BindConverter<T>)converter;
+            }
+
+            private static BindConverter<T> MakeTypeConverterConverter<T>()
+            {
+                var typeConverter = TypeDescriptor.GetConverter(typeof(T));
+                if (typeConverter == null || !typeConverter.CanConvertFrom(typeof(string)))
+                {
+                    throw new InvalidOperationException(
+                        $"The type '{typeof(T).FullName}' does not have an associated {typeof(TypeConverter).Name} that supports " +
+                        $"conversion from a string. " +
+                        $"Apply '{typeof(TypeConverterAttribute).Name}' to the type to register a converter.");
+                }
+
+                return ConvertWithTypeConverter;
+
+                bool ConvertWithTypeConverter(object obj, out T value)
+                {
+                    var text = (string)obj;
+                    if (string.IsNullOrEmpty(text))
+                    {
+                        value = default;
+                        return true;
+                    }
+
+                    // We intentionally close-over the TypeConverter to cache it. The TypeDescriptor infrastructure is slow.
+                    var converted = typeConverter.ConvertFromString(context: null, CultureInfo.InvariantCulture, text);
```

Comment:
Why use `CultureInfo.InvariantCulture` here? The other parsing cases use `CultureInfo.CurrentCulture` - shouldn't the typeconverter be allowed to be culture-sensitive too?